crates/core/src/surfnet/locker.rs

```diff
@@ -1164,17 +1186,36 @@ impl SurfnetSvmLocker {
         let signature = meta.signature;
 
         if do_propagate {
-            let meta = convert_transaction_metadata_from_canonical(&meta);
+            let meta_c = convert_transaction_metadata_from_canonical(&meta);
             let simnet_events_tx = self.simnet_events_tx();
             let _ = simnet_events_tx.try_send(SimnetEvent::error(format!(
                 "Transaction execution failed: {}",
                 err
             )));
-
             let _ = status_tx.try_send(TransactionStatusEvent::ExecutionFailure((
                 err.clone(),
-                meta,
+                meta_c,
             )));
+
+            let fee = 5000 * transaction.signatures.len() as u64;
```

Comment:
This calculation of the fee is logic that should be in the `from_failure` function, not outside of it